src/mpyl/steps/postdeploy/cypress_test.py

```diff
@@ -33,28 +36,70 @@ def execute(self, step_input: Input) -> Output:
                                 task_name="Verifying cypress")
+            execute_with_stream(logger=self._logger, container=docker_container, command="yarn tsc",
+                                task_name="Compiling typescript")
 
-            run_command = f"yarn cypress run --spec {specs_string}"
+            run_command = f'bash -c "yarn cypress run --spec {specs_string} || true"'
             record_key = cypress_config.record_key
-            if not step_input.run_properties.local and record_key:
-                run_command += f" --record --key {record_key}"
-            execute_with_stream(logger=self._logger, container=docker_container, command=run_command,
-                                task_name="Running cypress tests")
+            if record_key:
+                run_command = f'bash -c "yarn cypress run --spec {specs_string} --record --key ' \
+                              f'b6a2aab1-0b80-4ca0-a56c-1c8d98a8189c || true "'
+            result = execute_with_stream(logger=self._logger, container=docker_container, command=run_command,
+                                         task_name="Running cypress tests")
+
+            for stdout in result:
+                if record_key and "Recorded Run" in stdout:
+                    cypress_results_url = stdout.rstrip().rsplit('Recorded Run: ', 1)[1]
+                if "error Command failed with exit code" in stdout:
+                    raise DockerException(command_launched=[run_command], return_code=1)
+        except DockerException:
+            return Output(success=False,
+                          message=f"Cypress tests for project {step_input.project.name} have one or more failures",
+                          produced_artifact=input_to_artifact(artifact_type=ArtifactType.JUNIT_TESTS,
+                                                              step_input=step_input,
+                                                              spec={TEST_OUTPUT_PATH_KEY: volume_path,
+                                                                    "cypress_results_url": cypress_results_url}))
         finally:
             docker_container.stop()
             docker_container.remove()
 
-        return Output(success=True, message=f"Cypress tests for {step_input.project.name} passed",
-                      produced_artifact=None)
+        return Output(success=True, message=f"Cypress tests for project {step_input.project.name} passed",
+                      produced_artifact=input_to_artifact(artifact_type=ArtifactType.JUNIT_TESTS, step_input=step_input,
+                                                          spec={TEST_OUTPUT_PATH_KEY: volume_path,
+                                                                "cypress_results_url": cypress_results_url}))
+
+    @staticmethod
+    def _target_to_test_target(target: Target) -> str:
+        if target == Target.PULL_REQUEST_BASE:
```

Comment:
Don't you need PULL_REQUEST here as well?

Reply:
It's the default return value below

Reply:
indeed